spec/editor-linter-spec.js

```diff
@@ -43,26 +96,50 @@ describe('EditorLinter', function() {
   })
 
   describe('onShouldLint', function() {
-    it('is triggered on save', async function() {
-      let timesTriggered = 0
-      function waitForShouldLint() {
-        // Register on the textEditor
-        const editorLinter = new EditorLinter(textEditor)
-        // Trigger a (async) save
-        textEditor.save()
-        return new Promise((resolve) => {
-          editorLinter.onShouldLint(() => {
-            timesTriggered++
-            // Dispose of the current registration as it is finished
-            editorLinter.dispose()
-            resolve()
-          })
+    let timesTriggered
+    let editor
+
+    function waitForShouldLint() {
+      // Register on the textEditor
+      const editorLinter = new EditorLinter(editor)
+      return new Promise((resolve) => {
+        editorLinter.onShouldLint(() => {
+          timesTriggered++
+          // Dispose of the current registration as it is finished
+          editorLinter.dispose()
+          resolve()
         })
-      }
+      })
+    }
+
+    beforeEach(() => {
+      timesTriggered = 0
+    })
+
+    it('is triggered on save', async function() {
+      editor = textEditor
       expect(timesTriggered).toBe(0)
+      // Trigger an async save
+      textEditor.save()
       await waitForShouldLint()
+      textEditor.save()
       await waitForShouldLint()
       expect(timesTriggered).toBe(2)
     })
+
+    it('is triggered on reload', async function() {
+      const tempPath = await copyFileToTempDir(path.join(__dirname, 'fixtures', 'file.txt'))
+      const tempDir = path.dirname(tempPath)
+      editor = await atom.workspace.open(tempPath)
+
+      expect(timesTriggered).toBe(0)
+      modifyFile(tempPath)
+      await waitForShouldLint()
+      modifyFile(tempPath)
+      await waitForShouldLint()
+      expect(timesTriggered).toBe(2)
+
+      rimraf.sync(tempDir)
+    })
```

Comment:
Can you add a third parameter options here and specify a timeout of 1.5 seconds?

Reply:
I can try that, but when SSHing in and trying different debugging steps the test is timing out on the very first call to `waitForShouldLint`.

Reply:
I guess trying doesn't hurt 😛